core/engine.go

```diff
@@ -92,42 +93,78 @@ func NewEngine(
 		ExecutionScheduler: ex,
 		executionState:     ex.GetState(),
 
-		Options:        opts,
+		options:        opts,
 		runtimeOptions: rtOpts,
 		outputs:        outputs,
 		Metrics:        make(map[string]*stats.Metric),
 		Samples:        make(chan stats.SampleContainer, opts.MetricSamplesBufferSize.Int64),
 		stopChan:       make(chan struct{}),
 		logger:         logger.WithField("component", "engine"),
+		registry:       registry,
 		builtinMetrics: builtinMetrics,
 	}
 
-	e.thresholds = opts.Thresholds
-	e.submetrics = make(map[string][]*stats.Submetric)
-	for name := range e.thresholds {
-		if !strings.Contains(name, "{") {
-			continue
+	if !(e.runtimeOptions.NoSummary.Bool && e.runtimeOptions.NoThresholds.Bool) {
+		err := e.initSubMetricsAndThresholds()
+		if err != nil {
+			return nil, err
 		}
+	}
+
+	return e, nil
+}
 
-		parent, sm := stats.NewSubmetric(name)
-		e.submetrics[parent] = append(e.submetrics[parent], sm)
+func (e *Engine) getOrInitPotentialSubmetric(name string) (*stats.Metric, error) {
```

Comment:
I would suggest removing the "Potential" part of the name, and explicit the behavior of the function in its documentation instead.

Reply:
resolved by https://github.com/grafana/k6/pull/2426/commits/d7bc04b33223b6c00f944c6d57b82bdba4bef52f